reconcile/status_page_components.py

```diff
@@ -0,0 +1,261 @@
+        if desired.group_name:
+            group_id = self.group_name_to_id.get(desired.group_name, None)
+            if not group_id:
+                raise ValueError(f"Group {desired.group_name} referenced "
+                                 f"by {desired.name} does not exist")
+
+        # Special handling if a component needs to be moved out of any grouping
+        # We would need to use the component_group endpoint but for now let's
+        # just raise this as an error because of lazyness へ‿(ツ)‿ㄏ
+        if current and current.group_name and not desired.group_name:
+            raise ValueError(f"Remove grouping from the component "
+                             f"{desired.group_name} is currently unsupported")
+
+        component_update = dict(
+            name=desired.display_name,
+            description=desired.description
+        )
+        if group_id:
+            component_update["group_id"] = group_id
+
+        if current:
+            LOG.info(f"update component {desired.name}: {component_update}")
+            if not dry_run:
+                self._update_component(current.id, component_update)
+            return current.id
+        else:
+            LOG.info(f"create component {desired.name}: {component_update}")
+            if not dry_run:
+                return self._create_component(component_update)
+            else:
+                return None
+
+    def _update_component(self, id: str, data: dict[str, Any]) -> None:
+        self._client().components.update(id, **data)
+
+    def _create_component(self, data: dict[str, Any]) -> Optional[str]:
+        result = self._client().components.create(**data)
+        return result.get("id")
+
+    def delete_component(self, dry_run: bool, id: str) -> None:
+        if not dry_run:
+            self._client().components.delete(id)
+            self._rebuild_state()
+
+    def _fetch_components(self) -> list[AtlassianComponent]:
+        raw_components = self._client().components.list()
+        group_ids_to_name = {g.id: g.name for g in raw_components if g.group}
+        return [
+            AtlassianComponent(
+                **c.toDict(),
+                group_name=group_ids_to_name.get(c.group_id, None)
+            )
+            for c in raw_components
+        ]
+
+    def _client(self):
+        return statuspageio.Client(api_key=self.token,
+                                   page_id=self.page_id,
+                                   organization_id="unset")
+
+
+def fetch_pages() -> list[StatusPage]:
+    return [StatusPage(**p) for p in queries.get_status_pages()]
+
+
+def get_state() -> State:
+    settings = queries.get_app_interface_settings()
+    accounts = queries.get_aws_accounts()
+    return State(integration=QONTRACT_INTEGRATION,
+                 accounts=accounts,
+                 settings=settings)
+
+
+def run(dry_run: bool = False):
+    state = get_state()
+    status_pages = fetch_pages()
+
+    page_reconcile_error_occured = False
```

Comment:
`error` is fine IMO. It's also more consistent with the naming in other integrations.

Reply:
fixed